subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/support/KotlinCompiler.kt

```diff
@@ -146,12 +146,19 @@ fun compileTo(
                 addJvmClasspathRoot(kotlinStdlibJar)
             }
             val environment = kotlinCoreEnvironmentFor(configuration, disposable)
-            return compileBunchOfSources(environment)
+            return compileKotlin(environment)
         }
     }
 }
 
 
+private
+fun compileKotlin(environment: KotlinCoreEnvironment): Boolean {
+    org.jetbrains.kotlin.cli.common.environment.setIdeaIoUseFallback()
+    return compileBunchOfSources(environment)
```

Comment:
Is there a way to statically disable this after this is called? Do you want to reset the static state back?